src/Castle.Core/DynamicProxy/Generators/Emitters/MethodEmitter.cs

```diff
@@ -159,6 +159,18 @@ private void DefineParameters(ParameterInfo[] parameters)
 #if DOTNET45
 				if (parameter.HasDefaultValue && parameter.DefaultValue != null)
 				{
+					if (parameter.ParameterType == typeof(decimal) || parameter.ParameterType == typeof(decimal?))
+					{
+						/*
+						 * A decimal value may not be passed to SetConstant(), so omit the call
+						 * in this instance.
+						 * 
+						 * See https://github.com/castleproject/Core/issues/87 and https://bit.ly/1eXx6fF 

```

Comment:
I know the background on the issue, but it'd be nicer if the comment was a bit clearer that this is a defect/omission in .NET so in the future when the .NET defect is fixed someone can remove this code without worry in case our GitHub issues history isn't around anymore.

Could you also include the full URL for the MSDN link rather than a bit.ly link:
https://msdn.microsoft.com/en-au/library/system.reflection.emit.parameterbuilder.setconstant(v=vs.110).aspx